### ISSUE TITLE
SSViewer is removing single backslashes (\) unless escaped (\\)

### ISSUE THREAD

**Issue by hailwood** [silverstripe/silverstripe-framework#5831]
Basically as the title says,
I'm pretty sure this is new as we've used regular expressions in our Javascript before and don't recall the \ getting stripped.

Running Silverstripe 3.4

**Page.php**

````php
class Page_Controller extends Content_Controller {

    protected static $allowed_actions = ['test'];

    public function test(){
        return $this->renderWith('TestFile');
    }

}
````

**TestFile.ss**
````html
<script>
    var exp = /\s+/g;
</script>

<h1>Testing \a \ normal \\ string</h1>
````

**Expected output**
````html
<script>
    var exp = /\s+/g;
</script>

<h1>Testing \a \ normal \\ string</h1>
````

**Actual output**
````html
<script>
    var exp = /s+/g;
</script>

<h1>Testing a  normal \ string</h1>
````

**Comment by dhensby** [comment]
not new - duplicate of #1507 and explained here: https://github.com/silverstripe/silverstripe-framework/issues/1507#issuecomment-15863676

tl;dr of it is: this is intentional, SS templates are not HTML, so back-slashes have special meaning - present since 3.0
